### PR TITLE
refactor(pipeline): refactor pipeline report service to provider

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
refactor pipeline report service to provider

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=314659&iterationID=1297&pId=319600&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    refactor pipeline report service to provider           |
| 🇨🇳 中文    |      pipeline report pb + provider 改造        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chengjoey** [comment]
image

File: internal/tools/pipeline/aop/plugins/pipeline/apitest_report/plugin.go [hunk 1]
```diff
@@ -184,12 +186,13 @@ func (p *provider) Handle(ctx *aoptypes.TuneContext) error {
 	var reqMeta map[string]interface{}
 	b, _ := json.Marshal(reportMeta)
 	_ = json.Unmarshal(b, &reqMeta)
+	pbMeta, _ := structpb.NewStruct(reqMeta)
```
Comment (sfwn) [inline internal/tools/pipeline/aop/plugins/pipeline/apitest_report/plugin.go:189]: handle error, and json error above

File: internal/tools/pipeline/aop/plugins/pipeline/basic/plugin.go [hunk 2]
```diff
@@ -76,10 +80,16 @@ func (p *provider) Handle(ctx *aoptypes.TuneContext) error {
 		content.TaskInfos = append(content.TaskInfos, taskReport)
 	}
 
-	_, err = ctx.SDK.Report.Create(apistructs.PipelineReportCreateRequest{
+	var reqMeta map[string]interface{}
+	b, _ := json.Marshal(map[string]interface{}{
+		"data": content,
+	})
+	_ = json.Unmarshal(b, &reqMeta)
+	pbMeta, err := structpb.NewStruct(reqMeta)
```
Comment (sfwn) [inline internal/tools/pipeline/aop/plugins/pipeline/basic/plugin.go:88]: err not handled

File: internal/tools/pipeline/aop/plugins/pipeline/testplan_before/plugin.go [hunk 3]
```diff
@@ -54,10 +57,11 @@ func (p *provider) Handle(ctx *aoptypes.TuneContext) error {
 	meta[autotest.CmsCfgKeyDisplayName] = ctx.SDK.Pipeline.Snapshot.Secrets[autotest.CmsCfgKeyDisplayName]
 
 	// report
-	_, err := ctx.SDK.Report.Create(apistructs.PipelineReportCreateRequest{
+	pbMeta, _ := structpb.NewStruct(meta)
```
Comment (sfwn) [inline internal/tools/pipeline/aop/plugins/pipeline/testplan_before/plugin.go:60]: abstract a method

File: internal/tools/pipeline/spec/pipeline_report.go [hunk 4]
```diff
@@ -35,3 +39,17 @@ type PipelineReport struct {
 func (*PipelineReport) TableName() string {
 	return "dice_pipeline_reports"
 }
+
+func (p *PipelineReport) ConvertToPB() *pb.PipelineReport {
+	meta, _ := structpb.NewStruct(p.Meta)
```
Comment (sfwn) [inline internal/tools/pipeline/spec/pipeline_report.go:44]: how about error

File: apistructs/pipeline_report.go [hunk 5]
```diff
@@ -58,67 +55,7 @@ var (
...
-	PipelineIDs []uint64             `schema:"-"`
-	Sources     []PipelineSource     `schema:"source"`
-	Types       []PipelineReportType `schema:"type"`
-
-	/////////////////////////
-	// pipeline 分页查询参数 //
-	/////////////////////////
-	// labels
-	// &mustMatchLabel=key2=value3
-	MustMatchLabelsQueryParams []string `schema:"mustMatchLabel"`
```
Comment (sfwn) [inline apistructs/pipeline_report.go:100]: how to handle `schema` flag

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/pipeline_report.go (hunk 5 above)
Comment (chengjoey) [reply apistructs/pipeline_report.go:100]: image
image
`PageListPipelines` func will handle `schema` flag

File: internal/tools/pipeline/spec/pipeline_report.go (hunk 4 above)
Comment (chengjoey) [reply internal/tools/pipeline/spec/pipeline_report.go:44]: adjusted, handle the error

File: internal/tools/pipeline/aop/plugins/pipeline/basic/plugin.go (hunk 2 above)
Comment (chengjoey) [reply internal/tools/pipeline/aop/plugins/pipeline/basic/plugin.go]: image

report provider unified convert to pb meta

**Pull request merged**